### PR TITLE
Build flat dependency trees while auditing Gradle projects

### PR DESCRIPTION
**Pull request by asafgabai**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
A new, more memory-efficient structure for the dependency trees of Gradle projects.

### CODE REVIEW

File: xray/commands/audit/sca/java/javautils.go [hunk 1]
```diff
@@ -173,3 +176,73 @@ func (dm *dependencyMultimap) putChild(parent string, child *buildinfo.Dependenc
 func (dm *dependencyMultimap) getChildren(parent string) map[string]*buildinfo.Dependency {
 	return dm.multimap[parent]
 }
+
+// The structure of a dependency tree of a module in a Gradle/Maven project, as created by the gradle-dep-tree and maven-dep-tree plugins.
+type moduleDepTree struct {
+	Root  string                 `json:"root"`
+	Nodes map[string]depTreeNode `json:"nodes"`
```
Comment (yahavi) [inline xray/commands/audit/sca/java/javautils.go:183]: Let's add "omitempty" to ensure we don't fail projects without children
```suggestion
	Nodes map[string]depTreeNode `json:"nodes,omitempty"`
```

File: xray/commands/audit/sca/java/javautils.go [hunk 2]
```diff
@@ -173,3 +176,73 @@ func (dm *dependencyMultimap) putChild(parent string, child *buildinfo.Dependenc
...
+}
+
+type depTreeNode struct {
+	Children []string `json:"children"`
```
Comment (yahavi) [inline xray/commands/audit/sca/java/javautils.go:187]:
```suggestion
	Children []string `json:"children,omitempty"`
```

File: xray/commands/audit/sca/java/javautils.go [hunk 3]
```diff
@@ -173,3 +176,73 @@ func (dm *dependencyMultimap) putChild(parent string, child *buildinfo.Dependenc
...
+	for _, childId := range moduleTree.Nodes[currNodeId].Children {
+		childGav := GavPackageTypeIdentifier + childId
+		childNode := &xrayUtils.GraphNode{
+			Id:     childGav,
+			Nodes:  []*xrayUtils.GraphNode{},
+			Parent: currNode,
+		}
+		if currNode.NodeHasLoop() {
+			return
+		}
```
Comment (yahavi) [inline xray/commands/audit/sca/java/javautils.go:220]: Let's put it in the beginning of the loop

**Review by yahavi — CHANGES_REQUESTED** [review]
Top-level @asafgabai!
Please consider my inline comments.

File: xray/commands/audit/sca/java/javautils.go (hunk 1 above)
Comment (asafgabai) [reply xray/commands/audit/sca/java/javautils.go:183]: It's not necessary to use `omitempty` in this context, as `omitempty` is typically used when encoding Go objects to JSON to omit empty fields. In our case, we only parse the plugins' JSON results into Go objects, where missing fields are parsed as having their empty value.

File: xray/commands/audit/sca/java/javautils.go (hunk 3 above)
Comment (asafgabai) [reply xray/commands/audit/sca/java/javautils.go:220]: I moved it outside of the loop.

**Pull request merged**